paasta_tools/cli/cmds/spark_run.py

```diff
@@ -503,6 +505,19 @@ def sanitize_container_name(container_name):
     return re.sub("[^a-zA-Z0-9_.-]", "_", re.sub("^[^a-zA-Z0-9]+", "", container_name))
 
 
+@lru_cache(maxsize=1)
+def _get_docker_binary() -> str:
+    docker_binary = shutil.which("docker")
+    if not docker_binary:
+        raise RuntimeError("Unable to locate the 'docker' executable in PATH")
+    return docker_binary
```

Comment:
might be nice to toss this in utils.py or somewhere more importable since we more or less have the same logic in other places (e.g., in local-run)